sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -263,6 +266,20 @@ private static void addTimeSpanToSerializedSpans(TimeSpan span, List<Map<String,
     spans.add(spanMap);
   }
 
+  private static void deleteCurrentSessionFile(final @NotNull SentryOptions options, boolean isSync) {
+    if (!isSync) {
+      try {
+        options.getExecutorService().submit(() -> {
+          deleteCurrentSessionFile(options);
+        });
+      } catch (Throwable e) {
+        options.getLogger().log(WARNING, "Submission of deletion of the current session file rejected.", e);
```

Comment:
if there is an exception you can even decide to call `deleteCurrentSessionFile` here in the catch clause, if it is important

Reply:
If there is an error, this should be ended and cleaned up on the next app start, but we this deletion should avoid https://github.com/getsentry/sentry-java/blob/6de823e05405663927ff66a16cfc5ab645529cc1/sentry/src/main/java/io/sentry/PreviousSessionFinalizer.java#L74